tests/User_selenium_unittest.py

```diff
@@ -0,0 +1,126 @@
+        driver.get(self.base_url + "login")
+        # Good practice to clear the entry before typing
+        driver.find_element_by_id("id_username").clear()
+        # These credentials will be used by Travis when testing new PRs
+        # They will not work when testing on your own build
+        # Be sure to change them before submitting a PR
+        driver.find_element_by_id("id_username").send_keys(os.environ['DD_ADMIN_USER'])
+        driver.find_element_by_id("id_password").clear()
+        driver.find_element_by_id("id_password").send_keys(os.environ['DD_ADMIN_PASSWORD'])
+        # "Click" the but the login button
+        driver.find_element_by_css_selector("button.btn.btn-success").click()
+        return driver
+
+    def test_create_user(self):
+        # Login to the site.
+        driver = self.login_page()
+        # Navigate to the User managegement page
+        driver.get(self.base_url + "user")
+        # "Click" the dropdown button to see options
+        driver.find_element_by_id("dropdownMenu1").click()
+        # "Click" the add prodcut button
+        driver.find_element_by_link_text("New User").click()
+        # Fill in the Necessary User Details
+        # username, first name, last name, email, and permissions
+        # Don't forget to clear before inserting
+        # username
+        driver.find_element_by_id("id_username").clear()
+        driver.find_element_by_id("id_username").send_keys("propersahm")
+        # First Name
+        driver.find_element_by_id("id_first_name").clear()
+        driver.find_element_by_id("id_first_name").send_keys("Proper")
+        # Last Name
+        driver.find_element_by_id("id_last_name").clear()
+        driver.find_element_by_id("id_last_name").send_keys("Samuel")
+        # Email Address
+        driver.find_element_by_id("id_email").clear()
+        driver.find_element_by_id("id_email").send_keys("propersam@example.com")
+        # Select user with super user permissions
+        driver.find_element_by_name("is_superuser").click()  # Clicking will mark the checkbox
+        # "Click" the submit button to complete the transaction
+        driver.find_element_by_css_selector("input.btn.btn-primary").click()
+        # Query the site to determine if the user has been created
+        productTxt = driver.find_element_by_tag_name("BODY").text
+        # Assert ot the query to dtermine status of failure
+        self.assertTrue(re.search(r'User added successfully, you may edit if necessary.', productTxt) or
+            re.search(r'A user with that username already exists.', productTxt))
+
+    def test_user_edit_permissions(self):
+        # Login to the site. Password will have to be modified
+        # to match an admin password in your own container
+        driver = self.login_page()
+        # Navigate to User Management page
+        driver.get(self.base_url + "user")
+        # Select A user to edit
+        driver.find_element_by_link_text("Edit").click()
```

Comment:
This edits the admin user instead of the newly created one. changing admins permissions has some strange effects. For example, logging in does not redirect past the login page, but does not show a message for incorrect credentials either.

Reply:
This is now resolved. A specific user is filtered out before edit button is clicked